### PR TITLE
[MINOR][DOC] Fix style in examples across documentation

### PR DESCRIPTION
**Pull request by HyukjinKwon**
## What changes were proposed in this pull request?

This PR fixes the documentation as below:
-  Python has 4 spaces and Java and Scala has 2 spaces (See https://cwiki.apache.org/confluence/display/SPARK/Spark+Code+Style+Guide).
- Avoid excessive parentheses and curly braces for anonymous functions. (See https://github.com/databricks/scala-style-guide#anonymous)
## How was this patch tested?

N/A

### CODE REVIEW

**Comment by HyukjinKwon** [comment]
BTW, this is not fixing some wrong examples and inconsistent indentation codes in `structured-streaming-programming-guide.md` because https://github.com/apache/spark/pull/14564 is handling them. I made a separate PR for this because that PR is originally about fixing codes in `./examples`.

**Comment by HyukjinKwon** [comment]
Also just for reviewers, the inconsistent stuffs I listed in the PR description happen randomly across documentation. So, this fixes them to be consistent according to style guide lines and resembling the majority in documentation.

File: docs/graphx-programming-guide.md [hunk 1]
```diff
@@ -420,24 +421,24 @@ val defaultUser = ("John Doe", "Missing")
 val graph = Graph(users, relationships, defaultUser)
 // Notice that there is a user 0 (for which we have no information) connected to users
 // 4 (peter) and 5 (franklin).
-graph.triplets.map(
-    triplet => triplet.srcAttr._1 + " is the " + triplet.attr + " of " + triplet.dstAttr._1
-  ).collect.foreach(println(_))
+graph.triplets.map {
```
Comment (srowen) [inline docs/graphx-programming-guide.md]: I don't agree with these changes. A scope is not required here. It was correct as-is.

**Comment by srowen** [comment]
Although I agree with the whitespace changes, they won't even affect the rendered docs, right? I am not sure I'd make this much change for no visible upside. It's something I think I'd change as we go and edit the text over time.

**Comment by HyukjinKwon** [comment]
@srowen Thank you for review! I can revert non-visible ones but will just leave the visible ones. 

I am okay with reverting this (anyway now we know there are nits of spacing in documentation!)

File: docs/streaming-programming-guide.md [hunk 2]
```diff
@@ -1142,12 +1142,12 @@ val joinedStream = windowedStream.transform { rdd => rdd.join(dataset) }
 JavaPairRDD<String, String> dataset = ...
 JavaPairDStream<String, String> windowedStream = stream.window(Durations.seconds(20));
 JavaPairDStream<String, String> joinedStream = windowedStream.transform(
-    new Function<JavaRDD<Tuple2<String, String>>, JavaRDD<Tuple2<String, String>>>() {
-        @Override 
-        public JavaRDD<Tuple2<String, String>> call(JavaRDD<Tuple2<String, String>> rdd) {
-            return rdd.join(dataset);
-        }
+  new Function<JavaRDD<Tuple2<String, String>>, JavaRDD<Tuple2<String, String>>>() {
```
Comment (HyukjinKwon) [inline docs/streaming-programming-guide.md]: This change looks a bit a lot but I just changed 4 spaces to 2 spaces.

File: docs/graphx-programming-guide.md [hunk 3]
```diff
@@ -235,8 +235,9 @@ triplet view of a graph to render a collection of strings describing relationshi
 val graph: Graph[(String, String), String] // Constructed from above
 // Use the triplets view to create an RDD of facts.
 val facts: RDD[String] =
-  graph.triplets.map(triplet =>
-    triplet.srcAttr._1 + " is the " + triplet.attr + " of " + triplet.dstAttr._1)
+  graph.triplets.map { triplet =>
```
Comment (srowen) [inline docs/graphx-programming-guide.md]: The brace change should be reverted here to match the other examples.
The rest seems OK as it's at least visible, and makes style more consistent.

Comment (HyukjinKwon) [inline docs/graphx-programming-guide.md]: Sure, thank you.

**Comment by srowen** [comment]
LGTM

**Comment by srowen** [comment]
Merged to master

**Pull request closed without merging**